### PR TITLE
[translation] add error state tests

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves https://github.com/Azure/azure-sdk-for-python/issues/17653

### CODE REVIEW

File: sdk/documenttranslation/azure-ai-documenttranslation/tests/test_cancel_job.py [hunk 1]
```diff
@@ -15,9 +16,9 @@ class TestCancelJob(DocumentTranslationTest):
 
     @DocumentTranslationPreparer()
     @DocumentTranslationClientPreparer()
-    def test_list_statuses(self, client):
+    def test_cancel_job(self, client):
         # prepare containers and test data
-        blob_data = [b'This is some text']
+        blob_data = [Document(data=b'This is some text')]
```
Comment (maririos) [inline sdk/documenttranslation/azure-ai-documenttranslation/tests/test_cancel_job.py:21]: not related to your PR but how do u guarantee that the job is still running before you call cancel?

File: sdk/documenttranslation/azure-ai-documenttranslation/tests/test_translation.py [hunk 2]
```diff
@@ -156,4 +156,168 @@ def test_single_source_single_target_with_suffix(self, client):
...
+                targets=[
+                    TranslationTarget(
+                        target_url=target_container_sas_url,
+                        language_code="es"
+                    )
+                ]
+            )
+        ]
+
+        with pytest.raises(HttpResponseError) as e:
```
Comment (maririos) [inline sdk/documenttranslation/azure-ai-documenttranslation/tests/test_translation.py:180]: does Python check that error code is returned and things like that?

File: sdk/documenttranslation/azure-ai-documenttranslation/tests/test_translation.py [hunk 3]
```diff
@@ -156,4 +156,168 @@ def test_single_source_single_target_with_suffix(self, client):
...
+        ]
+
+        job = client.create_translation_job(translation_inputs)
+        job = client.wait_until_done(job.id)
+        self._validate_translation_job(job, status="Failed", total=1, failed=1)
+
+        doc_status = client.list_all_document_statuses(job.id)
+        doc = next(doc_status)
+        assert doc.status == "Failed"
+        assert doc.error.code == "TargetFileAlreadyExists"
```
Comment (maririos) [inline sdk/documenttranslation/azure-ai-documenttranslation/tests/test_translation.py:263]: yay!!! :P

**Review by maririos — COMMENTED** [review]
Error tests LGTM!

File: sdk/documenttranslation/azure-ai-documenttranslation/tests/test_cancel_job.py (hunk 1 above)
Comment (kristapratico) [reply sdk/documenttranslation/azure-ai-documenttranslation/tests/test_cancel_job.py:21]: Good point. Looking at the recordings, this was in a "NotStarted" state before it got canceled. Might make sense to add a few extra documents into the input to make sure this doesn't finish too quick

File: sdk/documenttranslation/azure-ai-documenttranslation/tests/test_translation.py (hunk 2 above)
Comment (kristapratico) [reply sdk/documenttranslation/azure-ai-documenttranslation/tests/test_translation.py:180]: This currently doesn't raise. Once the work for #17344 is in then I'll update it.

**Review by maririos — APPROVED** [review]

**Pull request merged**